### PR TITLE
fix: Consistent wasm builder

### PR DESCRIPTION
**Pull request by rbialek**
## Why?

Ensure that builds executed locally and on github produce identical wasm files.

## How?

- Use the same docker canister on github and locally
- Created fleek/f3o build canister with node,rust,dfx toolset 
- Updated build.yml workflow to use the canister
- Created docker-build.sh tool to generate local builds using the flee/f3o container

## Contribution checklist?

- [x] The commit messages are detailed
- [x] It does not break existing features (unless required)
- [x] I have performed a self-review of my own code
- [ ] Documentation has been updated to reflect the changes

## Demo?

Run local build script: `GithubActionPlugin/dockers/docker-build.sh`
```
Original:          1.15 MiB
Stripping Unused Data Segments...
    Size:          615.06 KiB (47.6% smaller)
Execute a binaryen optimization pass on your WASM....
    Size:          564.65 KiB (8.2% smaller)

Final Size: 564.65 KiB (51.9% smaller)
⚠️  The ic-cdk-optimizer completed!
🤖 Finished: Services builder
.dfx
`-- local
    `-- canisters
        |-- cover
        |   |-- cover.did
        |   |-- cover.did.d.ts
        |   |-- cover.did.js
        |   |-- cover.wasm
        |   `-- index.js
        `-- idl

4 directories, 5 files
Build completed in /home/rbialek/RubymineProjects/fleek/cover/dfx-build
```
Now you can check the checksum of the generated `cover.wasm`
```
GithubActionPlugin/scripts/checksum-wasm.sh dfx-build/local/canisters/cover/cover.wasm
0x28c9055a9f847a54812c11b12ec7172fc0c007c9a0e36160d9767b9119a97426
```

### CODE REVIEW

**Review by nicolaspoggi — APPROVED** [review]
readme lgtm, noted to update docs when merged

File: .docker/deployment/Dockerfile [hunk 1]
```diff
@@ -4,13 +4,15 @@
 #  -e CANISTERIUM_CANISTER_ID="rrkah-fqaaa-aaaaa-aaaaq-cai" \
 #  -e PAT=YOUR_PERSONAL_AT \
 #  fleek/f3o-fe
-FROM fleek/f3o:latest
+FROM fleek/f3o:0.2.2
```
Comment (scott-dn) [inline .docker/deployment/Dockerfile:7]: consider have a consistent name with the project

File: README.md [hunk 2]
```diff
@@ -80,6 +64,25 @@ Whenever you push your code using `production` or `main` branches, the above wor
...
 
+### Build Canister
+
+In order to get the same wasm files on github actions and locally, 
+we need to ensure that the build environment on github actions is EXACTLY the same as the local one. 
+Thus, if you want to generate a wasm file locally, you must use the same docker image as the github actions is using.
+
+You can either provide your own docker image (We suggest you use ubuntu:20:04 at the base) or 
+you use our fleek/f3o docker image that includes tools needed to build Rust based canisters. 
+The fleek/f3o image is build with this [Dockerfile](./GithubActionPlugin/dockers/f3o/Dockerfile).
```
Comment (scott-dn) [inline README.md:75]: shouldn't name directory with f3o

File: .docker/deployment/Dockerfile (hunk 1 above)
Comment (scott-dn) [reply .docker/deployment/Dockerfile:7]: also the `deployment` directory name doesn't seem to match with context

Comment (rbialek) [reply .docker/deployment/Dockerfile]: I'ts been renamed to fleek/dfxrust image

**Review by scott-dn — APPROVED** [review]
LGTM

**Pull request merged**